app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/BaseRepositoryImpl.java

```diff
@@ -146,7 +146,10 @@ public Flux<T> findAll(Example example) {
     public Mono<T> archive(T entity) {
         Assert.notNull(entity, "The given entity must not be null!");
         Assert.notNull(entity.getId(), "The given entity's id must not be null!");
-        Assert.isTrue(!entity.isDeleted(), "The given entity is already deleted");
+        // Entity is already deleted
+        if (entity.isDeleted()) {
+            return Mono.empty();
```

Comment:
Can we return the given entity as is, instead of empty Mono please? It'll be useful for further `.map` or `.flatMap` transformations.

```suggestion
            return Mono.just(entity);
```

Reply:
@sharat87 done